gluoncv/utils/metrics/segmentation.py

```diff
@@ -88,7 +103,12 @@ def batch_pix_accuracy(output, target):
 def batch_intersection_union(output, target, nclass):
     """mIoU"""
     # inputs are NDarray, output 4D, target 3D
+    # add new axis if missing batch dimension
     # the category -1 is ignored class, typically for background / boundary
+    if len(output.shape) == 3:
```

Comment:
`batch_intersection_union` is supposed to work for a batch with 4D.

Reply:
When running 
`python test.py --dataset ade20k --pretrained --model fcn --backbone resnet50 --eval`

The args shape for the function is 
 output: (150, 480, 640)
 target: (480, 640)
![Screenshot from 2019-10-17 20-22-55](https://user-images.githubusercontent.com/18246639/67114965-f84f2880-f191-11e9-9910-f03ed2b84fc5.png)

Caused by
https://github.com/dmlc/gluon-cv/blob/a78965fe0f4d6003a1bbd7a9553f6d71205daa5c/gluoncv/utils/metrics/segmentation.py#L43-L47
The labels here is a list of 2D, which passes label=2D into batch_pix_accuracy 
The preds here is a list of 3D, which passes pred=3D into batch_pix_accuracy

Reply:
I'll try to fix from the dataset

Reply:
OK I'll try to modify from the dataset